src/bot/middleware/only-with-text.middleware.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * @description
+ * Skip messages without text
+ * */
+function onlyWithText(ctx, next) {
+  /**
+   * Skip messages before bot became admin
+   * */
+  if ((ctx.msg?.date || 0) * 1000 < +ctx.session.botAdminDate) {
+    return;
+  }
+
+  if (ctx.msg.text || ctx.msg.caption) {
+    ctx.msg.text = ctx.msg.text || ctx.msg.caption;
```

Comment:
I don't think it will be very transparent for the developers. They may expect that text could sometimes be missing.

I suggest you store the message into the `ctx.state.text = ctx.msg.text || ctx.msg.caption`

```suggestion
    ctx.state.text = ctx.msg.text || ctx.msg.caption;
```

And then use `ctx.state.text` into the on-text.middleware